test/fields/field.cpp

```diff
@@ -11,6 +11,342 @@
 #include "NeoFOAM/fields/operations/operationsMacros.hpp"
 #include "NeoFOAM/fields/operations/comparison.hpp"
 
+TEST_CASE("Field Copy Constructor")
+{
+    SECTION("CPU")
+    {
+        int N = 10;
+        NeoFOAM::CPUExecutor cpuExec {};
+
+        NeoFOAM::Field<NeoFOAM::scalar> a(cpuExec, N);
+        NeoFOAM::fill(a, 5.0);
+
+        NeoFOAM::Field<NeoFOAM::scalar> b(a);
+
+        REQUIRE(b.size() == N);
+
+        for (int i = 0; i < N; i++)
+        {
+            REQUIRE(b.span()[i] == 5.0);
+        }
+    };
+
+    SECTION("OpenMP")
+    {
+        int N = 10;
+        NeoFOAM::OMPExecutor OMPExec {};
+
+        NeoFOAM::Field<NeoFOAM::scalar> a(OMPExec, N);
+        NeoFOAM::fill(a, 5.0);
+
+        NeoFOAM::Field<NeoFOAM::scalar> b(a);
+
+        REQUIRE(b.size() == N);
+
+        for (int i = 0; i < N; i++)
+        {
+            REQUIRE(b.span()[i] == 5.0);
+        }
+    };
+
+    SECTION("GPU")
+    {
+        int N = 10;
+        NeoFOAM::GPUExecutor gpuExec {};
+
+        NeoFOAM::Field<NeoFOAM::scalar> a(gpuExec, N);
+        NeoFOAM::fill(a, 5.0);
+
+        NeoFOAM::Field<NeoFOAM::scalar> b(a);
+
+        REQUIRE(b.size() == N);
+
+        for (int i = 0; i < N; i++)
+        {
+            REQUIRE(b.copyToHost().span()[i] == 5.0);
+        }
```

Comment:
I think this should also work.
```suggestion
       auto hostB = b.copyToHost();
        for (auto value: hostB.span())
        {
            REQUIRE( value == 5.0);
        }
```